packages/trie/src/trie/trie.ts

```diff
@@ -730,6 +737,175 @@ export class Trie {
     )
   }
 
+  /**
+   * Return a range proof of trie items between startingHash and limitHash
+   * This proof can be verified using `verifyRangeProof`.
+   * Note, that the proof will always include a proof for the startingHash (even if it does not exist)
+   * The limitHash could not be proved,
+   * @param startingHash This is the starting (hash) key item (note on non-hashed tries this is just a key)
+   * @param limitHash This is the limit (hash) key item (note on non-hashed tries this is just a key)
+   */
+  async createRangeProof(startingHash: Buffer, limitHash: Buffer) {
```

Comment:
TODO probably add a limit to how much we want to return also (devp2p/snap compat)

Reply:
Can we give an example here, what a "range" actually *is* in the context of a range proof here? So maybe referencing a super simple trie and give an example start and end and say what keys would be "in range"?

Also not 100% clear to me here just from "reading" the API and comments: is `limitHash` included or not? Why not just call this `startX` and `endX`?

Another naming suggestion: since this *can* be non hashed keys (a naming case we also have in other parts of the libray) why not rather call this (I'll take my naming suggestions from above, but feel free to decide separately) `startKey` and `endKey`? This would be the more generic case covering both base and secure trie cases and the comment additions could also go away?

Generally, thanks a lot here, seems that you guys are really starting to understand the dephts of the trie library, makes me really happy! 🙂

Reply:
(so, I was e.g. first thinking: so range likely would mean if we have a trie with keys 1,3,5 and 7 a "range" with start 1 and 5 would likely be 1,3,5, but then I thought: no, this very likely isn't right and to simple and "range" must be defined somewhat differently being based on the structure of the trie)

Reply:
`limitHash` is included, but your name suggestions are great. Your example is right though, if you have keys 1,3,5,7 and you request range request 1-5, you get 1,3,5. (These are all ascending keys and all are included)

Reply:
+1 for startKey and endKey, its clear that way, and I think range directly refers to the actual key structure of the trie, and yes limit (endKey) should be including if there exits an endKey, else the key  before that.
similarly for startKey as per my understanding